eppo_client/configuration_store.py

```diff
@@ -1,33 +1,25 @@
 from typing import Dict, Optional, TypeVar, Generic
-from cachetools import LRUCache
 
 from eppo_client.read_write_lock import ReadWriteLock
 
 T = TypeVar("T")
 
 
 class ConfigurationStore(Generic[T]):
-    def __init__(self, max_size: int):
-        self.__cache: LRUCache = LRUCache(maxsize=max_size)
+    def __init__(self):
+        self.__cache: Dict[str, T] = {}
         self.__lock = ReadWriteLock()
 
     def get_configuration(self, key: str) -> Optional[T]:
-        try:
-            self.__lock.acquire_read()
-            return self.__cache[key]
-        except KeyError:
-            return None  # key does not exist
-        finally:
-            self.__lock.release_read()
+        with self.__lock.reader():
```

Comment:
We aren't worried about two readers - I thought the problem is not wanting to read during a write? If so perhaps one lock is sufficient

Reply:
We block writing when there are >= 1 readers until they are done.

See implementation [here](https://github.com/Eppo-exp/python-sdk/blob/94b597f29a0d68fd90e91d5759e8b21733878b2d/eppo_client/read_write_lock.py)

Reply:
Nice!!